### DISCUSSION TITLE
31.0后无法进入游戏 / Cannot enter the game after update 31.0

### DISCUSSION THREAD

**Discussion by Pik-4** [Pik-4/HsMod#181] [Announcements]
目前暂时无法确定炉石Unity的具体版本，因此配置UI可能无法使用。配置计划使用web进行修改，~~目前仍在开发中~~（没人写Web，无限期拖延，web api在 #122）

缓解方案：
1. 删除UnstrippedCorlib目录下所有未裁切的Unity开头的`.dll`文件 。
2. 移除`BepInEx/patchers`目录下所有`.dll`
3. 移除`BepInEx/plugins`目录下所有`.dll`
4. 添加`HsMod.dll`到`BepInEx/plugins`
5. 重新进入炉石即可。
6. 修改配置需要在游戏启动前，修改`BepInEx/config/HsMod.cfg`或在启用快捷键后，使用快捷键进行控制。如果配置文件异常，建议启动游戏前删除`BepInEx/config/HsMod.cfg`，等待启动游戏后重新生成新的`HsMod.cfg`再进行修改，修改时游戏需要关闭。
7. 如果不理解1-6，建议删除'BepInEx'，参考readme重新配置。
8. （不稳定，非必要不建议进行这一步操作）向UnstrippedCorlib添加`UnityEngine.IMGUIModule.dll`（从[unity](https://unity.bepinex.dev/libraries/2021.3.40.zip)中提取），并添加[BepInEx.ConfigurationManager.BepInEx5](https://github.com/BepInEx/BepInEx.ConfigurationManager/releases)进行配置控制（默认游戏内`F1`）。

The specific version of Hearthstone Unity is currently unknown, so the configuration UI may not be available. The configuration is planned to be modified using the web, which is still under development (no one writes web, so it is delayed indefinitely, web api on #122)

Mitigation:
1. Delete all `.dll` files which starting with Unity in the UnstrippedCorlib directory.
2. Remove all `.dll` in the `BepInEx/patchers` directory
3. Remove all `.dll` in the `BepInEx/plugins` directory
4. Add `HsMod.dll` to the `BepInEx/plugins` directory
5. Re-enter the game.
6. To modify the configuration, you need to modify `BepInEx/config/HsMod.cfg` before starting the game, or use shortcut keys to control after enabling shortcut keys. If the configuration file is abnormal, it is recommended to delete `BepInEx/config/HsMod.cfg` before starting the game, wait for the game to start and regenerate a new `HsMod.cfg` before modifying it. The game needs to be closed when modifying.
7. If you do not understand 1-6, it is recommended to delete 'BepInEx' and reconfigure according to the readme.
8. (Unstable, Not recommended unless necessary) Add `UnityEngine.IMGUIModule.dll` to UnstrippedCorlib (extracted from [unity](https://unity.bepinex.dev/libraries/2021.3.40.zip)), and add [BepInEx.ConfigurationManager.BepInEx5](https://github.com/BepInEx/BepInEx.ConfigurationManager/releases) for configuration control (default in-game `F1`).

**Comment by lskyzero** [comment]
没用，进游戏 HsMod.cfg自动重置，默认关闭，无法调用

**Comment by freechao123** [comment]
进天梯会卡死在后手硬币回合

**Comment by xinghuyun** [comment]
大佬nb。不过建议大家备份一下设置和皮肤文件，我的皮肤文件没了，变成了没改之前的原始文件。

**Comment by ma9527** [comment]
> 没用，进游戏 HsMod.cfg自动重置，默认关闭，无法调用

f4能显示运行，f5打不开咋办

**Comment by freechao123** [comment]
> 大佬nb。不过建议大家备份一下设置和皮肤文件，我的皮肤文件没了，变成了没改之前的原始文件。

未裁剪的有哪些呀，我删了昨天那一个以后，天梯硬币卡死

**Comment by Lofuuzi** [comment]
> > 没用，进游戏 HsMod.cfg自动重置，默认关闭，无法调用
> 
> f4能显示运行，f5打不开咋办

當然打不開，管理器內嵌調用介面都刪了

**Comment by xinghuyun** [comment]
> > > 没用，进游戏 HsMod.cfg自动重置，默认关闭，无法调用
> > 
> > 
> > f4能显示运行，f5打不开咋办
> 
> 當然打不開，管理器內嵌調用介面都刪了

F5是打不开的，需要去改配置文件。

**Comment by freechao123** [comment]
> > > > 没用，进游戏 HsMod.cfg自动重置，默认关闭，无法调用
> > > 
> > > 
> > > f4能显示运行，f5打不开咋办
> > 
> > 當然打不開，管理器內嵌調用介面都刪了
> 
> F5是打不开的，需要去改配置文件。

天梯正常玩吗

**Comment by 1297656202** [comment]
好像用处不大，F4显示运行但是没有实际效果

**Comment by ForceofwiII** [comment]
改配置文件就行了

**Comment by lskyzero** [comment]
> 改配置文件就行了

问题是你配置文件进游戏之后自动重置，然后就等于没生效

**Comment by 1297656202** [comment]
对，进入对局后马上就会重置配置的

**Comment by freechao123** [comment]
> 对，进入对局后马上就会重置配置的

UnstrippedCorlib目录下unity文件全删了吗

**Comment by freechao123** [comment]
> > 改配置文件就行了
> 
> 问题是你配置文件进游戏之后自动重置，然后就等于没生效

UnstrippedCorlib目录下unity文件全删了吗

**Comment by freechao123** [comment]
> 改配置文件就行了

UnstrippedCorlib目录下unity文件全删了吗

**Comment by ForceofwiII** [comment]
> > 改配置文件就行了
> 
> UnstrippedCorlib目录下unity文件全删了吗

直接cv楼主新发的就行了

**Comment by lskyzero** [comment]
> > 改配置文件就行了
> 
> UnstrippedCorlib目录下unity文件全删了吗

早删了，就是照着做也不行才说的

**Comment by he1lluv** [comment]
> > > 改配置文件就行了
> > 
> > 
> > UnstrippedCorlib目录下unity文件全删了吗
> 
> 早删了，就是照着做也不行才说的

我只用金卡功能并没有出现重置

**Comment by 1297656202** [comment]
确实是可以用，直接把更新之后作者包里面的hsmod/UnstrippedCorlib的所有文件复制到对应地方就好，把hsmod.dll粘贴到原来的地方，然后就手动改hsmod.cfg文件即可，其他的文件都要像作者说的那样删除掉

**Comment by peterpanzj** [comment]
@Pik-4 最新更新的是删好的嘛，可以直接用吗

**Comment by yonota** [comment]
希望参与项目中，web方面我可以来写，如果可以请联系我

**Comment by ssdawesf** [comment]
> 确实是可以用，直接把更新之后作者包里面的hsmod/UnstrippedCorlib的所有文件复制到对应地方就好，把hsmod.dll粘贴到原来的地方，然后就手动改hsmod.cfg文件即可，其他的文件都要像作者说的那样删除掉

怎么修改hsmod.cfg  修改哪里？  我想用加速和金卡其他的不太用 谢谢了

**Comment by 1297656202** [comment]
> @Pik-4 最新更新的是删好的嘛，可以直接用吗

是的，看我说的

**Comment by peterpanzj** [comment]
> 希望参与项目中，web方面我可以来写，如果可以请联系我

哇可以可以+1需要帮助直接说，可以无偿帮忙

**Comment by DarkSissi** [comment]
> @Pik-4 最新更新的是删好的嘛，可以直接用吗

按照指导，把untiy开头的删除了，还有两个文件夹里的都删除了再用新的dll

**Comment by peterpanzj** [comment]
@Pik-4 需要帮忙说一声，web可以一起解决

**Comment by LoverD321** [comment]
我用VS COde修改HsMod.cfg文件没用  用pycharm竟然好使

**Comment by freechao123** [comment]
> 我用VS COde修改HsMod.cfg文件没用  用pycharm竟然好使

直接记事本不就能改了吗

**Comment by LoverD321** [comment]
> > 我用VS COde修改HsMod.cfg文件没用  用pycharm竟然好使
> 
> 直接记事本不就能改了吗

一开始记事本我试了也不行  也有可能是我不知道干啥了正好好使了

**Comment by yonota** [comment]
> > > 我用VS COde修改HsMod.cfg文件没用  用pycharm竟然好使
> > 
> > 
> > 直接记事本不就能改了吗
> 
> 一开始记事本我试了也不行 也有可能是我不知道干啥了正好好使了

文件只读或者没有给vscode和记事本权限，pycharm启动的时候可能默认请求了权限

**Comment by abevol** [comment]
> > 一开始记事本我试了也不行 也有可能是我不知道干啥了正好好使了
> 
> 文件只读或者没有给vscode和记事本权限，pycharm启动的时候可能默认请求了权限

我感觉是他在记事本里同时修改了语言的配置，导致 HsMod 重新生成了新的语言配置，导致他的其它修改全失效了🤣
然后换了新的编辑器后，就可以修改重新生成的配置了

**Comment by yaleiyaleicbt** [comment]
删了文件之后再想修改配置在哪里改啊

**Comment by z-henry** [comment]
是不是unityengine的dll都用炉石就行了

**Comment by 1297656202** [comment]
> > 确实是可以用，直接把更新之后作者包里面的hsmod/UnstrippedCorlib的所有文件复制到对应地方就好，把hsmod.dll粘贴到原来的地方，然后就手动改hsmod.cfg文件即可，其他的文件都要像作者说的那样删除掉
> 
> 怎么修改hsmod.cfg 修改哪里？ 我想用加速和金卡其他的不太用 谢谢了

Speed Gear Status = true 和Golden Card Effects改一个你平常用的，Shortcut Status = true，这样方便快捷键调节

**Comment by w229288802** [comment]
一打开游戏，配置就重置了

**Comment by Onlydukang1** [comment]
我只要后手有币就卡死了，这咋办啊？

**Comment by DarkSissi** [comment]
> 我只要后手有币就卡死了，这咋办啊？

没删文件。。。

缓解方案：

删除UnstrippedCorlib目录下所有未裁切的Unity开头的.dll文件 。
移除BepInEx/patchers目录下所有.dll
移除BepInEx/plugins目录下所有.dll
添加HsMod.dll到BepInEx/plugins
重新进入炉石即可。
修改配置需要在游戏启动前，修改BepInEx/config/HsMod.cfg或在启用快捷键后，使用快捷键进行控制。如果配置文件异常，建议启动游戏前删除BepInEx/config/HsMod.cfg，等待启动游戏后重新生成新的HsMod.cfg再进行修改，修改时游戏需要关闭。
如果不理解1-6，建议删除'BepInEx'，参考readme重新配置。

pik写的很清楚了，“未裁切的Unity开头的.dll文件”是指所有Unity开头的dll文件

**Comment by Roc1677313525** [comment]
号被封了14天。目前有风险，玩拨号贼慎用倍速。玩的拨号贼，大概玩了3个多小时，倍速5，其余没开。
D 16:00:46.2848891 Networ.ShowBreakingNewsOrError() - error: GLOBAL_ERROR_NETWORK_LOGIN_FAILURE
D 16:00:46.2848891 Network.ProcessDelayedError() - reason: UNKNOWN, message: 无法通过暴雪战网服务进行登录。请等待几分钟并再次尝试。
被封邮件
![2024-11-1被封](https://github.com/user-attachments/assets/344c70af-368e-4a96-91ec-f9e79a2ad19c)
被封前
[Hearthstone.log](https://github.com/user-attachments/files/17594137/Hearthstone.log)
失联
[Hearthstone.log](https://github.com/user-attachments/files/17594138/Hearthstone.log)

**Comment by abevol** [comment]
> 号被封了14天。目前有风险，玩拨号贼慎用倍速。玩的拨号贼，大概玩了3个多小时，倍速5，其余没开。

肯定是被对面举报了

**Comment by w229288802** [comment]
一打开游戏，加速配置就重置了

**Comment by w229288802** [comment]
哪位大佬帮忙解决一下

**Comment by Pik-4** [comment]
> 号被封了14天。目前有风险，玩拨号贼慎用倍速。玩的拨号贼，大概玩了3个多小时，倍速5，其余没开。 

正常来说第一次是封3天，另外，我这几天都2倍速兄弟，一点事没有

**Comment by lycxxz** [comment]
> > 确实是可以用，直接把更新之后作者包里面的hsmod/UnstrippedCorlib的所有文件复制到对应地方就好，把hsmod.dll粘贴到原来的地方，然后就手动改hsmod.cfg文件即可，其他的文件都要像作者说的那样删除掉
> 
> 感谢，如果小白不理解的话直接删掉旧的BepInEx，按照最新的英语版安装教程重新走一遍就好，最新的已经删除好了的不用删来删去了。之后手动去BepInEx/config的HsMod.cfg里手动修改一些配置就可以了。（需要先启动游戏才会生成config文件，config生成成功后退出游戏设置完再启动游戏就可以了）

我BepInEx目录下没有config文件夹怎么办，只有core plugins和unstripped_corlib

**Comment by JenseYoung** [comment]
> > > 确实是可以用，直接把更新之后作者包里面的hsmod/UnstrippedCorlib的所有文件复制到对应地方就好，把hsmod.dll粘贴到原来的地方，然后就手动改hsmod.cfg文件即可，其他的文件都要像作者说的那样删除掉
>  
> 
> > 感谢，如果小白不理解的话直接删掉旧的BepInEx，按照最新的英语版安装教程重新走一遍就好，最新的已经删除好了的不用删来删去了。之后启动游戏，如果安装成功的话就会生成config文件，这个时候点F4是会有提示的（没提示没生成的话重新回去走一遍教程，详细看下载了什么，x86版本等详细对应上）。生成完config文件手动打开BepInEx/config的HsMod.cfg里修改一些配置就可以了。

> 
> 我BepInEx目录下没有config文件夹怎么办，只有core plugins和unstripped_corli

**Comment by wexiao17** [comment]
> > > 确实是可以用，直接把更新之后作者包里面的hsmod/UnstrippedCorlib的所有文件复制到对应地方就好，把hsmod.dll粘贴到原来的地方，然后就手动改hsmod.cfg文件即可，其他的文件都要像作者说的那样删除掉
> > 
> > 
> > 感谢，如果小白不理解的话直接删掉旧的BepInEx，按照最新的英语版安装教程重新走一遍就好，最新的已经删除好了的不用删来删去了。之后手动去BepInEx/config的HsMod.cfg里手动修改一些配置就可以了。（需要先启动游戏才会生成config文件，config生成成功后退出游戏设置完再启动游戏就可以了）
> 
> 我BepInEx目录下没有config文件夹怎么办，只有core plugins和unstripped_corlib

config是配置文件，一般运行游戏一次就会生成，如果没有，那大概率是搭建的有问题导致dll没生效

**Comment by tuskermanshu** [comment]
> 希望参与项目中，web方面我可以来写，如果可以请联系我

+1 可以支援web

**Comment by FengCHENluo** [comment]
> > > > 确实是可以用，直接把更新之后作者包里面的hsmod/UnstrippedCorlib的所有文件复制到对应地方就好，把hsmod.dll粘贴到原来的地方，然后就手动改hsmod.cfg文件即可，其他的文件都要像作者说的那样删除掉
> > > 
> > > 
> > > 感谢，如果小白不理解的话直接删掉旧的BepInEx，按照最新的英语版安装教程重新走一遍就好，最新的已经删除好了的不用删来删去了。之后手动去BepInEx/config的HsMod.cfg里手动修改一些配置就可以了。（需要先启动游戏才会生成config文件，config生成成功后退出游戏设置完再启动游戏就可以了）
> > 
> > 
> > 我BepInEx目录下没有config文件夹怎么办，只有core plugins和unstripped_corlib
> 
> config是配置文件，一般运行游戏一次就会生成，如果没有，那大概率是搭建的有问题导致dll没生效
请问需要修改些什么啊？我进入游戏之后按f4没有反应是正常的吗？

**Comment by Pik-4** [comment]
@tuskermanshu @peterpanzj @yonota @DarkSissi
如果你们对web开发感兴趣，可以参考 #122 ，随意发挥。

**Comment by detao666** [comment]
> 号被封了14天。目前有风险，玩拨号贼慎用倍速。玩的拨号贼，大概玩了3个多小时，倍速5，其余没开。 D 16:00:46.2848891 Networ.ShowBreakingNewsOrError() - error: GLOBAL_ERROR_NETWORK_LOGIN_FAILURE D 16:00:46.2848891 Network.ProcessDelayedError() - reason: UNKNOWN, message: 无法通过暴雪战网服务进行登录。请等待几分钟并再次尝试。 被封邮件 ![2024-11-1被封](https://private-user-images.githubusercontent.com/126319156/382138672-344c70af-368e-4a96-91ec-f9e79a2ad19c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3MDQ2NDYsIm5iZiI6MTczMDcwNDM0NiwicGF0aCI6Ii8xMjYzMTkxNTYvMzgyMTM4NjcyLTM0NGM3MGFmLTM2OGUtNGE5Ni05MWVjLWY5ZTc5YTJhZDE5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNFQwNzEyMjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMjgxMDcxNTZlNTNkNmUzM2IxNGViOGRmMTU3MWE0ZmNjY2FhNzVjM2MxYWFiZDRkMjYwY2ExNjhjYzBlODYwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2KOFHcFgbA6do6J3B9_RQThGqSjUi0WF15d4S-k5YtU) 被封前 [Hearthstone.log](https://github.com/user-attachments/files/17594137/Hearthstone.log) 失联 [Hearthstone.log](https://github.com/user-attachments/files/17594138/Hearthstone.log)

国服敏感吧 外服感觉一直没啥事

**Comment by fishking0721** [comment]
作者能把快捷键状态默认开启吗？我调整HsMod.cfg开启快捷键之后每次启动游戏就会重置到关闭状态。重装等等试过很多次。HsMod.cfg调成只读的话直接不加载插件了

**Comment by abevol** [comment]
> 作者能把快捷键状态默认开启吗？我调整HsMod.cfg开启快捷键之后每次启动游戏就会重置到关闭状态。重装等等试过很多次。HsMod.cfg调成只读的话直接不加载插件了

你应该上传一下你修改后的 HsMod.cfg，这样作者才好判断问题出在哪里。

**Comment by fishking0721** [comment]
[HsMod.zip](https://github.com/user-attachments/files/17691026/HsMod.zip)

这是我最后的config配置，每次我启动游戏之后设置都会重置到这个状态/我修改快捷键=true之后启动游戏返回config看又被覆盖为false了
